### PR TITLE
9.6.1 with pagination on Transaction/Block/Mosaic

### PR DESCRIPTION
**Pull request by rg911**
### 9.6.1 with pagination
----
#### Model property name changes
- **MetadataEntry**: senderPublicKey: string => sourceAddress: Address; targetPublicKey: string => targetAddress: Address
- **MultisigAccountGraphInfo**: multisigAccounts => multisigEntries
- **MultisigAccountInfo**: account: PublicAccount => accountAddress: Address; cosignatories: PublicAccount[] => cosignatoryAddresses: Address; multisigAccounts: PublicAccount[] => multisigAddresses: Address[]
- **BlockInfo / NewBlock**: beneficiaryPublicKey: PublicAccount | undefined => beneficiaryAddress: Address | undefined
- **MosaicId**: owner: PublicAccount => ownerAddress: Address
- **MosaicInfo**: owner: PublicAccount => ownerAddress: Address; height => startHeight 
- **NamespaceInfo**: owner: PublicAccount => ownerAddress: Address
- **ChainProperties**: harvestNetworkFeeSinkPublicKey => harvestNetworkFeeSinkAddress
- **MosaicNetworkProperties**: mosaicRentalFeeSinkPublicKey => mosaicRentalFeeSinkAddress
- **NamespaceNetworkProperties**: namespaceRentalFeeSinkPublicKey => namespaceRentalFeeSinkAddress
- **NetworkProperties**: publicKey => nemesisSignerPublicKey
- **BalanceChangeReceipt**: targetPublicAccount: PublicAccount => targetAddress: Address
- **BalanceTransferReceipt**: sender: PublicAccount => senderAddress: Address

#### Transactions property name changes
- **AccountMetadataTransaction**: targetPublicKey: string => targetAddress: UnresolvedAddress
- **MosaicMetadataTransaction**: targetPublicKey: string => targetAddress: UnresolvedAddress
- **NamespaceMetadataTransaction**: targetPublicKey: string => targetAddress: UnresolvedAddress
- **MultisigAccountModificationTransaction**: publicKeyAdditions: PublicAccount[] => addressAdditions: UnresolvedAddress[]; publicKeyDeletions: PublicAccount[] => addressDeletions: UnresolvedAddress[]
- **AggregateTransactionService**: cosignatories: string[] => cosignatories: Address[]

#### Other changes
- New types created: UnresolvedAddress, UnresolvedMosaicId
- Address format changed from 25 bytes to 24 bytes
- Added sanity check on the new address format: Only `A, I, Q, Y` allowed as the last char on a valid address
- MosaicId creation (from Nonce) changed to use publicKey to address.
- Added reservedfield2 in TransferTransaction
- Added 8 bytes **version** in cosignature (default: 0).

#### Pagination
- Removed 'get transactions' endpoints from AccountHttp and BlockHttp
- Added `search` endpoint to TransactionHttp, BlockHttp, and MosaicHttp.
- Added SearchCriteria interfaces. 
- **group** filter in `TransactionSearchCriteria` has been made mandatory since rest split `getTransaction` into `getConfirmedTransaction`, `getUnconfirmedTransaction` and `getPartialTransaction`. Also same as `search transaction` into search confirmed/unconfirmed/partial transactions
- Added **streamer** for the 3 new search endpoints (block, mosaic, transaction).

### CODE REVIEW

File: src/infrastructure/TransactionRepository.ts [hunk 1]
```diff
@@ -43,26 +44,13 @@ export interface TransactionRepository extends Searcher<Transaction, Transaction
...
-    getTransactionsStatuses(transactionHashes: string[]): Observable<TransactionStatus[]>;
-
     /**
      * Gets a transaction's effective paid fee
      * @param transactionId - Transaction id or hash.
+     * @param transactionGroup - Transaction group.
      * @returns Observable<number>
      */
-    getTransactionEffectiveFee(transactionId: string): Observable<number>;
+    getTransactionEffectiveFee(transactionId: string, transactionGroup: TransactionSearchGroup): Observable<number>;
```
Comment (fboucquez) [inline src/infrastructure/TransactionRepository.ts:53]: Do you need the group in the effective fee? should it always be confirmed?

File: src/infrastructure/TransactionSearchGroup.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A transaction could be classified in the following groups: * Unconfirmed: The transaction reached the P2P network. At this point, it is not guaranteed that the transaction will be included in a block. * Confirmed: The transaction is included in a block. * Partial: The transaction requires to be cosigned by other transaction participants in order to be included in a block. * Failed: The transaction did not pass the network validation, and it was rejected.
+ */
+export enum TransactionSearchGroup {
```
Comment (fboucquez) [inline src/infrastructure/TransactionSearchGroup.ts:20]: Can we rename it to `TransactionGroup`? It's going to be used in more ways than just search endpoint.

The way I see it is: 
TransactionGroup = confirmed | partial | unconfirmed
TransactionState =  confirmed | partial | unconfirmed | error

I've also added an optional group to the transaction in Java to be filled depending when how the transaction is loaded. If the transaction is loaded using the `confirmed` search, `confirmed` listener or `confirmed` by id, the group is `confirmed`. The state/group is not going to be added to the rest payload but it's still good that the user has the state/group at hand

https://github.com/nemtech/symbol-openapi/issues/159

https://github.com/nemtech/symbol-sdk-java/blob/961/sdk-core/src/main/java/io/nem/symbol/sdk/model/transaction/Transaction.java#L279

File: e2e/infrastructure/MultisigAccounts.spect.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
+/*
```
Comment (dgarcia360) [inline e2e/infrastructure/MultisigAccounts.spect.ts:1]: typo in the file name: spect

File: src/infrastructure/TransactionSearchGroup.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A transaction could be classified in the following groups: * Unconfirmed: The transaction reached the P2P network. At this point, it is not guaranteed that the transaction will be included in a block. * Confirmed: The transaction is included in a block. * Partial: The transaction requires to be cosigned by other transaction participants in order to be included in a block. * Failed: The transaction did not pass the network validation, and it was rejected.
```
Comment (dgarcia360) [inline src/infrastructure/TransactionSearchGroup.ts:18]: There is no failed group in the current enum  - might need to remove it from the comment.

File: src/infrastructure/searchCriteria/BlockSearchCriteria.ts [hunk 5]
```diff
@@ -30,7 +30,7 @@ export interface BlockSearchCriteria extends SearchCriteria {
     /**
      * beneficiary public key. (optional)
```
Comment (dgarcia360) [inline src/infrastructure/searchCriteria/BlockSearchCriteria.ts:31]: beneficiary public key -> beneficiary  address

**Pull request merged**